ClassLibrary/ClassLibrary/Library.cs

```diff
@@ -21,5 +21,65 @@ public class Library
 
         public Reader GetReader(int IDReader) => departmentReaders.GetReader(IDReader);
 
+        public void ReaderBoroweBook(Reader reader,Book book)
+        {
+            if (!departmentReaders.CheckReader(reader))
+            {
+                throw new ArgumentException($"Not Exist {reader.ArgumentsToString()} in DepartmentReaders");
+            }
+
+            if (repositoryBooks.GetBook(book.ID)==null)
+            {
+                throw new ArgumentException($"This book not exist {book.ArgumentsToString()} in RepositoryBooks");
+            }
+            if (repositoryBooks.CountBookWithISBN(book.ISBN)==departmentReaders.CountBorrowedBooksWithISBN(book.ISBN))
+            {
+                throw new ArgumentException($"ALL books with {book.ISBN} borrowed");
+            }
+
+
+            departmentReaders.BorrowBook(reader, book);
+        }
+
+        public void ReaderBoroweBook(int IDReader, String NameBook)
+        {
+            if (departmentReaders.GetReader(IDReader)==null)
+            {
+                throw new ArgumentException($"Not Exist Reader with (ID={IDReader}) in DepartmentReaders");
+            }
+
+            if (NameBook.IsNullOrWhiteSpace())
+            {
+                throw new ArgumentNullException($"Not set Namebook");
+            }
+
+            var listAllBookWithName = repositoryBooks.FindBooks(NameBook);
+          
+            if (listAllBookWithName.Count == 0)
+            {
+                    throw new ArgumentException($"This book ({NameBook}) not exist in RepositoryBooks");
+            }
+
+           String ISBN = listAllBookWithName[0].ISBN;
+
+           HashSet<Book> allBookWithISBN= new HashSet<Book>(listAllBookWithName);
```

Comment:
Why `HashSet`?

Reply:
because HashSet:
1) all books unique, it has ID property
2) I wanna take free books: I take all book with ISBN and remove borrowed book
List nasn't metthod, if not use predicate

-------
commit push late, while correct